### PR TITLE
Add schemas for WP REST API search results

### LINKED ISSUE #16: Add definition for REST API search results
* `/wp/v2/search`

### PR DESCRIPTION
**Pull request by kevinwhoffman**
## Description

Resolves #16

This PR includes the following:

- New schema: `WP_REST_API_Search_Result`
- New schema: `WP_REST_API_Search_Results` (which references `WP_REST_API_Search_Result`)
- Updated `schema.json` with new schemas
- Updated `readme.md` with new schemas

New schemas are formatted according to `CONTRIBUTING.md` and tested with `npm run test`.

### CODE REVIEW

File: schemas/rest-api/search-results.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+	"$schema": "http://json-schema.org/draft-07/hyper-schema#",
+	"$id": "https://raw.githubusercontent.com/johnbillion/wp-json-schemas/trunk/schemas/rest-api/search-results.json",
+	"title": "WP_REST_API_Search_Results",
+	"description": "A collection of search result objects in a REST API context.",
+	"type": "array",
+	"items": {
+		"$ref": "search-result.json"
+	},
+	"links": [
```
Comment (kevinwhoffman) [inline schemas/rest-api/search-results.json:10]: @johnbillion I wasn't quite sure how to handle the `links` portion of the schema, so please review and perhaps clarify how links should be handled in contributing guidelines. I did compare other schemas `links` to the items found in the `_links` portion of their corresponding `OPTIONS` request, however I found that the `OPTIONS` request often included more `_links` items than the schema did.

**Comment by johnbillion** [comment]
This looks good! I'm going to push some tests and then document how to add them. I haven't been too strict with tests and I should be.

File: schemas/rest-api/search-results.json (hunk 1 above)
Comment (johnbillion) [reply schemas/rest-api/search-results.json:10]: The `_links` property is something I haven't gotten around to adding full schema information for. The existing `object-links.json` schema describes the shape of any `_links` property, but not the actual link elements for any endpoints.

**Comment by kevinwhoffman** [comment]
Sounds good. Just let me know when you're done with docs updates and I'll be happy to add tests.

**Comment by johnbillion** [comment]
Here's the commit that added the tests: e32aa5f. Basically all the commands in `tests/mu-plugins/mu-plugin.php` contribute to the test data so there's no need to manually generate any. They all need to be filled out with more data to cover more variations in data but that can be done in separate PRs.

**Comment by johnbillion** [comment]
Can you update the docs in `packages/wp-types/readme.md` too? Then I think that's everything covered.

**Pull request merged**